sstables/mc/parquet_writer.hh

```diff
@@ -0,0 +1,286 @@
+    }
+
+private:
+
+
+    template<typename ValueType>
+    std::vector<ValueType> deserialize(const std::vector<bytes> &cell_vec, const column_definition &def) {
+        std::vector<ValueType> vec;
+        vec.reserve(cell_vec.size());
+        for (const auto &cell: cell_vec) {
+            const data_value value = def.type->deserialize_value(cell);
+            vec.push_back(value_cast<ValueType>(value));
+        }
+        return vec;
+    }
+
+    template<typename Writer, typename ValueType>
+    void write_column(seastarized::RowGroupWriter *rgw, const std::vector<bytes> &cells, const column_definition &def) {
+        std::vector<ValueType> deserialized_values = deserialize<ValueType>(cells, def);
+        Writer *col_writer = static_cast<Writer *>(rgw->NextColumn().get0());
+        col_writer->WriteBatch(deserialized_values.size(), nullptr, nullptr, deserialized_values.data()).get0();
+    }
+
+    void write_byte_array(seastarized::RowGroupWriter *rgw,
+                          const std::vector<bytes> &cells,
+                          const column_definition &def) {
+        seastarized::ByteArrayWriter *col_writer = static_cast<seastarized::ByteArrayWriter *>(rgw->NextColumn().get0());
+
+        for (const auto &cell: cells) {
+            // TODO this is probably bad, need to use deserialize_value
+            sstring str = def.type->to_string(cell);
+            parquet::ByteArray value;
+            value.ptr = (const uint8_t *) (str.c_str());
+            value.len = str.size();
+            int16_t definition_level = 1;
+            col_writer->WriteBatch(1, &definition_level, nullptr, &value).get();
+        }
+    }
+
+
+    void flush_buffer() {
+        // TODO split into row groups more accurately
+        //  now we write into one big row group
+        auto rgw = _pq_file_writer->AppendRowGroup().get0();
+
+        // In this iteration we assume that the ordering of columns
+        // in the schema has not changed after convert_to_parquet method
+        const auto &columns = _sstable_schema.all_columns();
+        for (const auto &col_def : columns) {
+            auto cells_it = _buffer.find(col_def.id);
+            auto type = map_column_type(col_def.type->get_kind());
+            if (cells_it == _buffer.end() || type == Type::UNDEFINED) {
+                continue;
+            }
+
+            switch (type) {
+
+                case Type::INT32:
+                    write_column<seastarized::Int32Writer, int32_t>(rgw, cells_it->second, col_def);
+                    break;
+                case Type::BYTE_ARRAY:
+                    write_byte_array(rgw, cells_it->second, col_def);
+                    break;
+                default:
+                    break;
+
+            }
+
+        }
+    }
+
+public:
+
+    void consume_end_of_stream() {
+        if (!is_test_ks()) return;
+        trace("consume_end_of_stream") << std::endl;
+        flush_buffer();
```

Comment:
Currently I flush the buffer only once I reach to the end of the stream. Obviously this is horribly wrong and we need to take into account the size of the buffer as well (not to get OOM).

Reply:
Indeed. In such cases, please put a `//FIXME: ` comment in the code, so that people are aware of it, e.g.:
```cpp
//FIXME: we need to flush the buffer more often if it gets too large in order to avoid OOM
flush_buffer();
```